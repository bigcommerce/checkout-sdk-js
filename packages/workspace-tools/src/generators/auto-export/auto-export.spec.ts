import path from 'path';

import autoExport from './auto-export';

describe('autoExport()', () => {
    it('export matching members from files to another file', async () => {
        const options = {
            inputPath: path.join(__dirname, '/__fixtures__/**/index.ts'),
            outputPath: path.join(__dirname, '/__temp__/output.ts'),
            memberPattern: '^Strategy',
        };

        expect(await autoExport(options)).toMatchSnapshot();
    });

    it('handles scenario where no matching member is found', async () => {
        const options = {
            inputPath: path.join(__dirname, '/__fixtures__/**/index.ts'),
            outputPath: path.join(__dirname, '/__temp__/output.ts'),
            memberPattern: '^Test',
        };

        expect(await autoExport(options)).toMatchSnapshot();
    });
});
