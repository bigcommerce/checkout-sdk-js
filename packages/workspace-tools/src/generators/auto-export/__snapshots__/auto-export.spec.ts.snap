// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`autoExport() export matching members from files to another file 1`] = `
"export { StrategyA } from '../__fixtures__/strategy-a';
export { StrategyB } from '../__fixtures__/strategy-b';
"
`;

exports[`autoExport() handles scenario where no matching member is found 1`] = `""`;
